key-share/src/valid.rs

```diff
@@ -5,8 +5,71 @@ use std::fmt;
 /// `Valid<T>` wraps a value `T` that has been validated using [`Validate`] trait.
 ///
 /// `Valid<T>` provides only immutable access to `T`. For instance, if you want to change content of `T`, you
-/// need to [deconstruct](Valid::into_inner) it, do necessary modifications, and then validate it again using `TryFrom`.
+/// need to [deconstruct](Valid::into_inner) it, do necessary modifications, and then validate it again.
+///
+/// ## Transitive "valideness" through `AsRef`
+/// `Valid<T>` assumes that if `T` implements `AsRef<K>` and `K` can be validated (i.e. `K` implements [`Validate`]),
+/// then `K` has been validated when `T` was validated. Thus, if you have value of type `Valid<T>`, you can obtain
+/// `&Valid<K>` via `AsRef` trait.
+///
+/// Example of transitive valideness is demostrated below:
```

Comment:
I'm not sure such a long example is necessary. for me this feature feels very natural. Instead I would just mention like "An example of transitive validness is obtaining `KeyInfo` from `CoreKeyShare`"

Reply:
The example I wrote is simplified all-contained version of these structs, it should be easier to read.